From d88047093bf523423ab0e378ab09ff1eeb9bd7cb Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 4 Aug 2016 23:59:24 -0400
Subject: [PATCH] raise the string length limit a little

50 page books with color codes can break the 32k limit and disconnect clients
---
 src/main/java/net/minecraft/server/PacketDataSerializer.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main/java/net/minecraft/server/PacketDataSerializer.java b/src/main/java/net/minecraft/server/PacketDataSerializer.java
index 4f1d7c2..104eeaf 100644
--- a/src/main/java/net/minecraft/server/PacketDataSerializer.java
+++ b/src/main/java/net/minecraft/server/PacketDataSerializer.java
@@ -306,8 +306,8 @@ public class PacketDataSerializer extends ByteBuf {
     public PacketDataSerializer a(String s) {
         byte[] abyte = s.getBytes(Charsets.UTF_8);
 
-        if (abyte.length > 32767) {
-            throw new EncoderException("String too big (was " + s.length() + " bytes encoded, max " + 32767 + ")");
+        if (abyte.length > 40767) { // EMC
+            throw new EncoderException("String too big (was " + s.length() + " bytes encoded, max " + 40767 + ")"); // EMC
         } else {
             this.d(abyte.length);
             this.writeBytes(abyte);
-- 
2.9.2

